src/chart/generateCategoricalChart.tsx

```diff
@@ -2280,6 +2280,19 @@ export const generateCategoricalChart = ({
               graphicalItem: { ...graphicalItem, childIndex },
               payload: graphicalItem.props.data[activeIndex],
             };
+          } else if (itemDisplayName === 'Pie') {
+            const tooltipPayload = activeItem.tooltipPayload?.[0].payload.payload;
+
+            const activePieIndex = item.props.data.findIndex((datum: typeof tooltipPayload) => {
+              return _.isEqual(tooltipPayload, datum);
+            });
```

Comment:
I believe the same holds true as above, this could also have duplicates, right? 
I wonder whether we can further refactor this logic a bit and possibly shared the computation of the activeIndex?

Reply:
Yes. I initially thought this wasn't possible, but I was incorrect. I've added a duplicate data item to the story. I've also refactored the logic to handle getting the active index and accounting for duplicates for multiple graph components.